File: go/day8_part1_2017.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

func main() {
	// Step 1: Read Input
	file, err := os.Open("input.txt")
	if err != nil {
		fmt.Println("File reading error", err)
		return
	}
	defer file.Close()

	// Step 2: Initialize Registers
	registers := make(map[string]int)

	// Step 3: Process Instructions
	scanner := bufio.NewScanner(file)
	for scanner.Scan() {
		parts := strings.Fields(scanner.Text())
		reg := parts[0]
		op := parts[1]
		amount, _ := strconv.Atoi(parts[2])
		condReg := parts[4]
		condOp := parts[5]
		condVal, _ := strconv.Atoi(parts[6])

		// Check condition
		cond := false
		switch condOp {
		case ">":
			cond = registers[condReg] > condVal
		case ">=":
			cond = registers[condReg] >= condVal
		case "<":
			cond = registers[condReg] < condVal
		case "<=":
			cond = registers[condReg] <= condVal
		case "==":
			cond = registers[condReg] == condVal
		case "!=":
			cond = registers[condReg] != condVal
		}

		if cond {
			switch op {
			case "inc":
				registers[reg] += amount
			case "dec":
				registers[reg] -= amount
			}
		}
	}

	// Step 4: Find Max Value
	maxValue := 0
	for _, value := range registers {
		if value > maxValue {
			maxValue = value
		}
	}

	fmt.Println(maxValue)
}

File: php/day8_part1_2017.php
<?php
// Step 1: Read Input
$input = file_get_contents("input.txt");

// Step 2: Initialize Registers
$registers = [];

// Step 3: Process Instructions
$lines = explode("\n", trim($input));
foreach ($lines as $line) {
    $parts = preg_split('/\s+/', $line);
    $reg = $parts[0];
    $op = $parts[1];
    $amount = (int)$parts[2];
    $condReg = $parts[4];
    $condOp = $parts[5];
    $condVal = (int)$parts[6];

    // Ensure registers are initialized
    if (!isset($registers[$reg])) {
        $registers[$reg] = 0;
    }
    if (!isset($registers[$condReg])) {
        $registers[$condReg] = 0;
    }

    // Check condition
    $cond = false;
    switch ($condOp) {
        case '>':
            $cond = $registers[$condReg] > $condVal;
            break;
        case '>=':
            $cond = $registers[$condReg] >= $condVal;
            break;
        case '<':
            $cond = $registers[$condReg] < $condVal;
            break;
        case '<=':
            $cond = $registers[$condReg] <= $condVal;
            break;
        case '==':
            $cond = $registers[$condReg] == $condVal;
            break;
        case '!=':
            $cond = $registers[$condReg] != $condVal;
            break;
    }

    if ($cond) {
        switch ($op) {
            case 'inc':
                $registers[$reg] += $amount;
                break;
            case 'dec':
                $registers[$reg] -= $amount;
                break;
        }
    }
}

// Step 4: Find Max Value
$maxValue = max($registers);

echo $maxValue;
?>